### ISSUE TITLE
There is a problem with mscash2-opencl

### ISSUE THREAD

**Issue by claudioandre** [magnumripper/JohnTheRipper#2206]
It hangs. See also: https://github.com/magnumripper/JohnTheRipper/pull/2203

Reproducible on super using:
```
$ ../run/john -test=0 -format=mscash2-opencl
Testing: mscash2-opencl, MS Cache Hash 2 (DCC2) [PBKDF2-SHA1 OpenCL]... Device 6: GeForce GTX TITAN X
```

**Comment by claudioandre** [comment]
I tried it on the AMD driver and:

```
Broadcast message from root@super.openwall.net
	(unknown) at 2:12 ...

The system is going down for reboot in 1 minute!
```

**Comment by claudioandre** [comment]
FYI
***
I saw PBKDF2-HMAC-SHA512-opencl stall on my last PR (the error disappeared when I asked for a rebuild).

I guess one recent commit introduced (or is triggering an ancient) an bug.

**Comment by magnumripper** [comment]
This is a new problem, right? But nothing changed in the format and I also doubt Super's driver was upgraded. So what could it be?

**Comment by claudioandre** [comment]
Stalls
$ git checkout a1bac03e412c4f7a4e0218f91298a5b7e8958241

Works
$ git checkout 4eb01693b225e2da1922585011aa3b944268511e
****

A new QA session would be a nice thing.

**Comment by claudioandre** [comment]
I mean mscash2 run stalls. I was only able to see a warning on pbkdf2_hmac_sha512.

**Comment by magnumripper** [comment]
This seems be have been caused by a1bac03 and fixed with f4e8ac0.

**Comment by claudioandre** [comment]
You guys have more (and better) hardware than I do; I'm going to PR mscash2 enabled on Travis.

Then, I will close the issue.
